plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/imports/ReferenceResolutionFinder.java

```diff
@@ -149,8 +151,10 @@ public void findResolutions(
 		boolean needCollisionCheck = !isUnresolvedReference && !isPropertyAccess;
 
 		List<IEObjectDescription> candidates = new ArrayList<>(512);
+		Map<IEObjectDescription, N4JSProjectConfigSnapshot> projects = new HashMap<>();
```

Comment:
Implementations of `IEObjectDescription` do not usually implement `#hashCode()`/`#equals()` (e.g. `ForwardingEObjectDescription`, `N4JSEObjectDescription`). Could `URI` be used as key here?

Reply:
Actually, I need the identity because later, the project is accessed for each candidate.